accounts/views.py

```diff
@@ -36,4 +37,61 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         user = self.object
         context["tweet_list"] = Tweet.objects.select_related("user").filter(user=user)
+        context["is_following"] = FriendShip.objects.filter(follower=self.request.user, following=user).exists()
+        context["following"] = FriendShip.objects.filter(follower=user).count()
+        context["follower"] = FriendShip.objects.filter(following=user).count()
+        return context
+
+
+class FollowView(LoginRequiredMixin, generic.RedirectView):
+    url = reverse_lazy("tweets:home")
+    http_method_names = ["post"]
+
+    def post(self, request, *args, **kwargs):
+        follower = self.request.user
+        following = get_object_or_404(User, username=self.kwargs["username"])
+        if self.kwargs["username"] == request.user.username:
+            return HttpResponseBadRequest("自分をフォローすることはできません。")
```

Comment:
```suggestion
        if self.kwargs["username"] == request.user.username:
            return HttpResponseBadRequest("自分をフォローすることはできません。")
        following = get_object_or_404(User, username=self.kwargs["username"])
```
UnFollowViewに合わせた方がここでは良さそうですね。理由は、request.user.usernameが一致した段階でreturnされるので、無駄にクエリが走らないようになりそうだからです。

とはいえどちらが先でもそこまで大差ないので、どちらもでもいいです。ただ、FollowView, UnFollowVIewどちらかに書き方は統一してください